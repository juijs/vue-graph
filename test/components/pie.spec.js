import { mount } from '@vue/test-utils'
import GraphPie from '@/components/pie'

describe('pie.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphPie, {
            props: {
                width: 500,
                height: 500,
                activeIndex: [ 0, 2 ],
                showTextType: 'inside',
                values: [ 10, 5, 5, 5 ],
                names: [ 'Apple', 'Google', 'MS', 'Facebook' ],
                dataFormat: function(a, b) {
                    if(b) return b + "%";
                    return a;
                }
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});