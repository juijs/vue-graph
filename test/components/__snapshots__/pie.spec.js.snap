// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pie.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="500"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="500"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="400"
            width="400"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="400"
            width="400"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="400"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="400"
        x="0"
        y="0"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="400"
          stroke="transparent"
          width="400"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-pie"
        transform="translate(50,50)"
      >
        <g
          transform="translate(204.75528258147577,198.45491502812527)"
        >
          <path
            d="M0,-200 A200,200 0 0,1 117.55705045849464,161.80339887498945 L0,0 Z"
            fill="#7977C2"
            opacity="1"
            stroke="#ececec"
            stroke-width="1"
          />
        </g>
        <g
          transform="translate(200,200)"
        >
          <path
            d="M117.55705045849464,161.80339887498945 A200,200 0 0,1 -117.55705045849459,161.80339887498948 L0,0 Z"
            fill="#7BBAE7"
            opacity="0.5"
            stroke="#ececec"
            stroke-width="1"
          />
        </g>
        <g
          transform="translate(195.24471741852423,201.54508497187473)"
        >
          <path
            d="M-117.5570504584946,161.80339887498948 A200,200 0 0,1 -190.2113032590307,-61.80339887498945 L0,0 Z"
            fill="#FFC000"
            opacity="1"
            stroke="#ececec"
            stroke-width="1"
          />
        </g>
        <g
          transform="translate(200,200)"
        >
          <path
            d="M-190.21130325903073,-61.80339887498945 A200,200 0 0,1 -4.973799150320701e-14,-200 L0,0 Z"
            fill="#FF7800"
            opacity="0.5"
            stroke="#ececec"
            stroke-width="1"
          />
        </g>
        <g
          visibility="visible"
        >
          <text
            fill="#333"
            font-size="11"
            opacity="1"
            text-anchor="middle"
            transform="translate(297.48329292025323,168.3257580765679)"
            y="3"
          >
            10%
          </text>
        </g>
        <g
          visibility="visible"
        >
          <text
            fill="#333"
            font-size="11"
            opacity="0.5"
            text-anchor="middle"
            transform="translate(200,300)"
            y="3"
          >
            5%
          </text>
        </g>
        <g
          visibility="visible"
        >
          <text
            fill="#333"
            font-size="11"
            opacity="1"
            text-anchor="middle"
            transform="translate(102.51670707974677,231.67424192343213)"
            y="3"
          >
            5%
          </text>
        </g>
        <g
          visibility="visible"
        >
          <text
            fill="#333"
            font-size="11"
            opacity="0.5"
            text-anchor="middle"
            transform="translate(141.22147477075268,119.09830056250527)"
            y="3"
          >
            5%
          </text>
        </g>
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
